app/design/frontend/base/default/template/boltpay/catalog/product/configure_checkout.phtml

```diff
@@ -61,18 +69,37 @@ var boltConfigPDP = {
      * @returns {boolean}
      */
     validate: function () {
-        if (this.product.stock.manage && (!this.product.stock.status || this.product.stock.qty < this.getQty())) {
-            if ((typeof BoltPopup !== 'undefined')) {
-                BoltPopup.setMessage('<?php echo $this->__("The requested quantity is not available."); ?>');
+        var formValidation = new Validation('product_addtocart_form');
+        try {
+            if (this.product.stock.manage && (!this.product.stock.status || this.product.stock.qty < this.getQty())) {
+                throw '<?php echo $this->__("The requested quantity is not available."); ?>';
+            }
+            //TODO: support file type custom options
+            //currently not supported because it would require custom upload and expiration logic
+            document.querySelectorAll("#product_addtocart_form input[type='file']").forEach(function(fileInput) {
+                if (fileInput.value) {
+                    var name = 'file-upload-not-supported-bolt';
+                    var advice = Validation.getAdvice(name, fileInput);
+                    if (advice == null) {
+                        advice = Validation.createAdvice(name, fileInput, false, '<?php echo $this->__('Files not supported with Product Page BoltCheckout. Please add product to cart and then use Bolt from the cart page to checkout.'); ?>');
+                    }
+                    Validation.showAdvice(fileInput, advice, name);
+                    fileInput.focus();
+                    throw '<?php echo $this->__('Files not supported with Product Page BoltCheckout. Please add product to cart and then use Bolt from the cart page to checkout.'); ?>';
+                }
+            });
+        } catch (e) {
+            if ((typeof BoltPopup !== 'undefined' && typeof e === 'string')) {
+                BoltPopup.setMessage(e);
                 BoltPopup.show();
             }
             return false;
         }
-        return new Validation('product_addtocart_form').validate();
+        return formValidation.validate();
     },
     /**
      * Returns cart data provided to BoltConnect.configureProductCheckout
-     * @returns {{total: number, items: ([{reference: number, quantity: number, price: number, name: string, options: string}]|[])}}
+     * @returns {{currency: string, items: ([{reference: number, quantity: number, price: number, name: string, options: string}]|[])}}
```

Comment:
Trying to understand this. Why was total removed?

Reply:
It is not required.  It should be a calculated value.  It is not spec'd, and also most alarmingly, core in SF seems to be changing this.  This has changed from a JS number to requiring a JS string.  We don't want to rely on this value that is unnecessary and removing it is easier and more reliable than having PPC break due to core changes.

Spec:
https://docs.bolt.com/docs/54-product-checkout

New error being raised:
![image](https://user-images.githubusercontent.com/973856/75790163-11382f00-5d6b-11ea-92d7-81a1cd4e928f.png)

Reply:
Got it.